<launch>

  <node name="line_detection_fu_node" pkg="line_detection_fu" type="line_detection_fu_node" > 
    <param name="cam_w"                         type="int"    value="640"/>  
    <param name="cam_h"                         type="int"    value="480"/> <!--we will use only bottom half of this...-->	
	<param name="proj_y_start"            	    type="int"	  value="50"/>
  	<param name="proj_image_h"            	    type="int" 	  value="160"/>
  	<param name="proj_image_w"            	    type="int" 	  value="160"/>
  	<param name="proj_image_horizontal_offset"  type="int"    value="0"/>
    <param name="roi_top_w"            	        type="int" 	  value="132"/>
  	<param name="roi_bottom_w"            	    type="int" 	  value="54"/>
    <param name="maxYRoi"                       type="int"    value="159"/>
    <param name="minYDefaultRoi"                type="int"    value="110"/>
    <param name="minYPolyRoi"                   type="int"    value="68"/>
    <param name="defaultXLeft"                  type="int"    value="20"/>
    <param name="defaultXCenter"                type="int"    value="60"/>
    <param name="defaultXRight"                 type="int"    value="100"/>
    <param name="interestDistancePoly"          type="int"    value="5"/>
    <param name="interestDistanceDefault"       type="int"    value="20"/>
    <param name="iterationsRansac"              type="int"    value="30"/>
    <param name="proportionThreshould"          type="double" value="0.6"/>
    <param name="m_gradientThreshold"           type="int"    value="8"/>
    <param name="m_nonMaxWidth"                 type="int"    value="10"/>
    <param name="laneMarkingSquaredThreshold"   type="int"    value="36"/>
    <param name="angleAdjacentLeg"              type="int"    value="18"/>
    <param name="scanlinesVerticalDistance"     type="int"    value="2"/>
    <param name="scanlinesMaxCount"             type="int"    value="100"/>
    <param name="polyY1"                        type="int"    value="155"/>
    <param name="polyY2"                        type="int"    value="145"/>    
    <param name="polyY3"                        type="int"    value="130"/>    
    <param name="detectLaneStartX"              type="int"    value="155"/> <!--this was originally 0.7m, which is about 90 in our case... used for gradient computation, might help us to get rid of too crazy polynomials-->
    <param name="maxAngleDiff"                  type="int"    value="999"/>

    <param name="camera_name"     		type="string"   value="app/camera/rgb/image_raw"/>      
    <param name="cam_deg"             	type="double"   value="8"/>
    <param name="cam_height"           	type="double"   value="18"/>


    <!-- jack 13 720p calibration: -->
    <!--param name="f_u"             	type="double"   value="927.054138"/>
    <param name="f_v"             		type="double"   value="935.428772"/>
    <param name="c_u"             		type="double"   value="654.456376"/>
    <param name="c_v"             		type="double"   value="346.241851"/-->
    

    <!-- jack 13 640*480 - cut to only bottom half (640*240) calibration: -->
    <param name="f_u"             		type="double"   value="655.554626"/>
    <param name="f_v"             		type="double"   value="652.052734"/>
    <param name="c_u"             		type="double"   value="312.773367"/>
    <param name="c_v"             		type="double"   value="7.779505"/>
    
  </node>
</launch>
